syntax = "proto3";

package test;

message EchoRequest {
  string text = 1;
}

message ComplexMessage {
  string _str = 1;
  float  _float = 2;
  repeated sint64 _int = 3;
  bytes  _bin = 4;
  bool   _bool = 5;
  message SubMessage {
    string field = 1;
    repeated EchoRequest echo = 2;
  }
  SubMessage sub = 10;
  enum SubEnum {
    VALUE1 = 0;
    VALUE2 = 20;
  };
  repeated SubEnum _enum = 11;
  repeated string str_list = 12;
}

// vim: sts=4 sw=4 et
